homeassistant/components/livisi/switch.py

```diff
@@ -23,54 +30,53 @@ async def async_setup_entry(
     """Set up switch device."""
     coordinator: LivisiDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
 
-    async def async_discover_device(device) -> None:
+    @callback
+    def handle_coordinator_update() -> None:
         """Add switch."""
-        livisi_switch: SwitchEntity = await create_entity(
-            hass, config_entry, device, coordinator
-        )
-        LOGGER.debug("Include device type: %s", device.get("type"))
-        async_add_entities([livisi_switch])
+        shc_devices: list[dict[str, Any]] = coordinator.data
+        for device in shc_devices:
+            if device["type"] == PSS_DEVICE_TYPE:
+                if device not in coordinator.switch_devices:
```

Comment:
Could we store a set with the id of each device instead? That may be more efficient.

Reply:
I agree, now we only store the id of the device,